### PR TITLE
Improved handling of integer time index

### PR DESCRIPTION
**Pull request by rwedge**
Addresses #99 and #125.

* Time values in `cutoff_time` that are not numeric or datetime will raise an error
* If target entity has an integer time index, feature matrix will also have integer time index
* Extra columns in `cutoff_time` will now pass through correctly when using integer time index
* Incompatible time index and cutoff time types will raise an error

### CODE REVIEW

File: featuretools/computational_backends/calculate_feature_matrix.py [hunk 1]
```diff
@@ -266,7 +277,10 @@ def calc_results(time_last, ids, precalculated_features=None, training_window=No
 
         # if all aggregations have been approximated, can calculate all together
         if no_unapproximated_aggs and approximate is not None:
-            grouped = [[datetime.now(), group]]
+            if isinstance(group[target_time].iloc[0], _numeric_types):
+                grouped = [[np.inf, group]]
+            else:
+                grouped = [[datetime.now(), group]]
```
Comment (kmax12) [inline featuretools/computational_backends/calculate_feature_matrix.py]: should we move this `datetime.now()` outside of `calc_results`. basically, we want every chunk that's part of a single cfm call to use the same time.

File: featuretools/entityset/entity.py [hunk 2]
```diff
@@ -591,25 +591,19 @@ def _filter_and_sort(self, df, time_last=None,
         """
         if self.time_index:
             if time_last is not None and not df.empty:
```
Comment (kmax12) [inline featuretools/entityset/entity.py]: is it possible for time last to be none by this point in the code? If not, let's remove

Comment (rwedge) [reply featuretools/entityset/entity.py]: `Entity.get_all_instances` and a few others call this without specifying a time last

File: featuretools/computational_backends/calculate_feature_matrix.py [hunk 3]
```diff
@@ -110,6 +113,11 @@ def calculate_feature_matrix(features, cutoff_time=None, instance_ids=None,
     if not isinstance(cutoff_time, pd.DataFrame):
         if cutoff_time is None:
             cutoff_time = datetime.now()
+            # if integer time index, use max value as cutoff time instead
+            if target_entity.time_index:
```
Comment (kmax12) [inline featuretools/computational_backends/calculate_feature_matrix.py]: the target entity might not have a time index, but some other entity in the entityset might. perhaps it makes sense to add a method or attribute to an entityset that returns if it is a datetime or numeric time index. 

we should assume all entities in an entityset are using the same units for the time index, so this method could do the check when being initialized or when `set_time_index` is called.

File: featuretools/variable_types/variable.py [hunk 4]
```diff
@@ -291,6 +291,11 @@ class TimeIndex(Variable):
     _dtype_repr = "time_index"
 
 
+class NumericTimeIndex(TimeIndex, Numeric):
```
Comment (kmax12) [inline featuretools/variable_types/variable.py]: can you add to API reference in documentation?

**Comment by kmax12** [comment]
Looks great. Merging

**Pull request merged**